### PR TITLE
[stdlib] Remove a bunch of unused public inlinable functions from Runtime

### PR DESCRIPTION
**Pull request by moiseev**
<rdar://problem/45746339>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please benchmark

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
@milseman can you take a look if this removes something UTF-8 uses?

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Review by milseman — CHANGES_REQUESTED** [review]
Pretty much all of these annotations should go away, not be approved for inclusion in our ABI

File: stdlib/public/core/Runtime.swift.gyb [hunk 1]
```diff
@@ -264,51 +191,51 @@ public func _swift_stdlib_atomicFetch${operation}Int(
 }
 % end
 
-@_fixed_layout // FIXME(sil-serialize-all)
+@_fixed_layout
 public final class _stdlib_AtomicInt {
```
[the remark this answers is not in the document]
Comment (milseman) [reply stdlib/public/core/Runtime.swift.gyb]: Agreed

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
Updated the PR to just removing the unused functions.

**Review by lorentey — APPROVED** [review]
The removals look good!

**Comment by moiseev** [comment]
@swift-ci Please test macOS platform

**Comment by moiseev** [comment]
Resolved the conflicts.
@swift-ci Please smoke test

**Review by milseman — APPROVED** [review]
LGTM

**Pull request merged**